### PR TITLE
removed task, added consumers for check existence of entities

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/907

### CODE REVIEW

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs [hunk 1]
```diff
@@ -13,88 +16,85 @@
...
 
-        //Todo rework this
-        public void Execute(AddUsersToProjectRequest request)
-        {
-            List<string> errors = new();
+        errors.Add(errorMessage);
+      }
+    }
+    public AddUsersToProjectCommand(
+        IUserRepository repository,
```
Comment (Cheephot) [inline src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs:55]: fix tabs

File: src/ProjectService.Data/UserRepository.cs [hunk 2]
```diff
@@ -1,176 +1,181 @@
...
+    public IEnumerable<DbProjectUser> Find(Guid userId)
+    {
+      return Find(new FindDbProjectsUserFilter { UserId = userId });
+    }
+
+    public IEnumerable<DbProjectUser> Find(FindDbProjectsUserFilter filter)
+    {
+      if (filter == null)
+      {
+        throw new ArgumentNullException(nameof(filter));
```
Comment (Cheephot) [inline src/ProjectService.Data/UserRepository.cs:76]: return null

File: src/ProjectService.Data/UserRepository.cs [hunk 3]
```diff
@@ -1,176 +1,181 @@
...
+
+      return CreateGetPredicates(filter, dbProjectsUser).ToList();
+    }
+
+    public bool AreUserProjectExist(Guid userId, Guid projectId, bool? isManager)
+    {
+      if (isManager.HasValue && isManager.Value)
+      {
+        return _provider
+            .ProjectsUsers
```
Comment (Cheephot) [inline src/ProjectService.Data/UserRepository.cs:89]: tab

File: src/ProjectService.Data/UserRepository.cs [hunk 4]
```diff
@@ -1,176 +1,181 @@
...
+            .Any(x => x.UserId == userId && x.ProjectId == projectId && x.Role == (int)ProjectUserRoleType.Manager && x.IsActive);
+      }
+
+      return _provider
+          .ProjectsUsers
```
Comment (Cheephot) [inline src/ProjectService.Data/UserRepository.cs:94]: tab

File: src/ProjectService.Data/UserRepository.cs [hunk 5]
```diff
@@ -1,176 +1,181 @@
...
 using LT.DigitalOffice.ProjectService.Models.Dto.Requests.Filters;
 using Microsoft.EntityFrameworkCore;
-using System;
-using System.Collections.Generic;
-using System.Linq;
 
 namespace LT.DigitalOffice.ProjectService.Data
 {
-    public class UserRepository : IUserRepository
+  public class UserRepository : IUserRepository
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Data/UserRepository.cs:15]: fixed

File: src/ProjectService.Data/UserRepository.cs [hunk 6]
```diff
@@ -1,176 +1,181 @@
...
+
+    public void AddUsersToProject(IEnumerable<DbProjectUser> dbProjectUsers, Guid projectId)
+    {
+      if (dbProjectUsers == null)
+      {
+        throw new ArgumentNullException(nameof(dbProjectUsers));
+      }
+
+      _provider.ProjectsUsers.AddRange(dbProjectUsers);
+      _provider.Save();
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Data/UserRepository.cs]: fixed

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs [hunk 7]
```diff
@@ -13,88 +16,85 @@
...
-                    _logger.LogWarning(logMessage, projectId, string.Join(", ", userIds));
-                    errors.Add(errorMessage);
-                }
-            }
-            catch (Exception exc)
-            {
-                _logger.LogError(exc, logMessage, projectId, string.Join(", ", userIds));
+      try
+      {
+        var response = _rcCreateWorkTime.GetResponse<IOperationResult<bool>>(
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs]: fixed

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs]: fixed

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs [hunk 8]
```diff
@@ -13,88 +16,85 @@
...
+      }
 
-            List<DbProjectUser> dbProjectUsers = request.Users.Select(user =>
-                GetDbProjectUsers(user, request.ProjectId)
-            ).ToList();
+      _validator.ValidateAndThrowCustom(request);
 
-            _repository.AddUsersToProject(dbProjectUsers, request.ProjectId);
+      List<DbProjectUser> dbProjectUsers = request.Users.Select(user =>
+          GetDbProjectUsers(user, request.ProjectId)
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs]: fixed

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20210819092000_UpdateDatabase.cs [hunk 9]
```diff
@@ -127,17 +127,17 @@ private void AddTaskPropertiesTable(MigrationBuilder migrationBuilder)
...
                 {
-                    nameof(DbTaskProperty.Id),
-                    nameof(DbTaskProperty.Name),
-                    nameof(DbTaskProperty.ProjectId),
-                    nameof(DbTaskProperty.CreatedBy),
-                    nameof(DbTaskProperty.PropertyType),
-                    nameof(DbTaskProperty.Description),
-                    nameof(DbTaskProperty.CreatedAtUtc),
-                    nameof(DbTaskProperty.IsActive)
+                    "Id",
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20210819092000_UpdateDatabase.cs]: fixed

File: src/ProjectService.Broker/DisactivateUserConsumer.cs [hunk 10]
```diff
@@ -17,7 +17,7 @@ public class DisactivateUserConsumer : IConsumer<IDisactivateUserRequest>
 
         public Task Consume(ConsumeContext<IDisactivateUserRequest> context)
         {
-            _repository.Remove(context.Message.UserId, context.Message.ModifiedBy);
+            _repository.RemoveAsync(context.Message.UserId, context.Message.ModifiedBy);
```
Comment (MariyaSin) [inline src/ProjectService.Broker/DisactivateUserConsumer.cs:20]: await?

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 11]
```diff
@@ -36,19 +36,18 @@ public class CreateImageCommand : ICreateImageCommand
     private readonly IDbImageMapper _dbProjectImageMapper;
     private readonly ICreateImageValidator _validator;
     private readonly IUserRepository _userRepository;
-    private readonly ITaskRepository _taskRepository;
 
     private List<Guid> CreateImages(List<ImageContent> context, Guid userId, Guid enityId, List<string> errors)
     {
-      List<CreateImageData> images = context
+      var images = context
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: do not use var

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 12]
```diff
@@ -36,19 +36,18 @@ public class CreateImageCommand : ICreateImageCommand
...
         .Select(x => new CreateImageData(x.Name, x.Content, x.Extension, userId))
         .ToList();
 
-      string logMessage = $"Errors while creating images for project id {enityId}.";
+      var logMessage = $"Errors while creating images for project id {enityId}.";
 
       try
       {
-        IOperationResult<ICreateImagesResponse> response = _rcImages.GetResponse<IOperationResult<ICreateImagesResponse>>(
+        var response = _rcImages.GetResponse<IOperationResult<ICreateImagesResponse>>(
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: do not use var

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 13]
```diff
@@ -114,7 +104,7 @@ public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
         };
       }
 
-      if (!_validator.ValidateCustom(request, out List<string> errors))
+      if (!_validator.ValidateCustom(request, out var errors))
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Image/CreateImageCommand.cs:108]: do not use var

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 14]
```diff
@@ -127,10 +117,10 @@ public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
 
       OperationResultResponse<List<Guid>> response = new();
 
-      List<Guid> imagesIds = CreateImages(
+      var imagesIds = CreateImages(
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Image/CreateImageCommand.cs:121]: do not use var

File: src/ProjectService.Business/Commands/Image/RemoveImageCommand.cs [hunk 15]
```diff
@@ -84,22 +82,14 @@ private bool RemoveImage(List<Guid> ids, List<string> errors)
...
-      {
-        task = _taskRepository.Get(request.EntityId, false);
-      }
-
       Guid userId = _httpContextAccessor.HttpContext.GetUserId();
-      if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects)
-        && !(request.ImageType == ImageType.Task && _userRepository.AreUserProjectExist(task.ProjectId, userId))
-        && !(request.ImageType == ImageType.Project && _userRepository.AreUserProjectExist(request.EntityId, userId, true)))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveProjects)
+        && !(request.ImageType == ImageType.Project && await _userRepository.DoesExistAsync(request.ProjectId, userId, true)))
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Image/RemoveImageCommand.cs:92]: i think, do not need to check imageType

File: src/ProjectService.Business/Commands/Project/FindProjectsCommand.cs [hunk 16]
```diff
@@ -34,8 +31,8 @@ public class FindProjectsCommand : IFindProjectsCommand
     private readonly IBaseFindFilterValidator _findFilterValidator;
     private readonly IFindProjectsResponseMapper _responseMapper;
     private readonly IRequestClient<IGetDepartmentsRequest> _rcGetDepartments;
-    private readonly IHttpContextAccessor _httpContextAccessor;
-    private readonly IConnectionMultiplexer _cache;
+    private readonly IRedisHelper _redisHelper;
+    private readonly IResponseCreater _responseCreater;
 
     private async Task<List<DepartmentData>> GetDepartments(List<Guid> departmentsIds, List<string> errors)
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/FindProjectsCommand.cs:37]: GetDepartmentsAsync

File: src/ProjectService.Business/Commands/Project/FindProjectsCommand.cs [hunk 17]
```diff
@@ -96,32 +93,26 @@ private async Task<List<DepartmentData>> GetDepartmentsThroughBroker(List<Guid>
...
       _findFilterValidator = findFilterValidator;
       _responseMapper = responseMapper;
       _rcGetDepartments = rcGetDepartments;
-      _httpContextAccessor = httpContextAccessor;
-      _cache = cache;
+      _redisHelper = redisHelper;
+      _responseCreater = responseCreater;
     }
 
     public async Task<FindResultResponse<ProjectInfo>> Execute(FindProjectsFilter filter)
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/FindProjectsCommand.cs:108]: ExecuteAsync

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 18]
```diff
@@ -344,7 +341,7 @@ public async Task<OperationResultResponse<ProjectResponse>> Execute(GetProjectFi
         List<ImageInfo> imagesInfos = await GetUserAvatars(usersDatas.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList(), response.Errors);
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:341]: GetUsersAvatarsAsync

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs [hunk 19]
```diff
@@ -1,19 +1,21 @@
...
 using LT.DigitalOffice.Kernel.Responses;
 using LT.DigitalOffice.Models.Broker.Requests.Time;
 using LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers.Interfaces;
 using LT.DigitalOffice.ProjectService.Data.Interfaces;
 using LT.DigitalOffice.ProjectService.Mappers.Db.Interfaces;
 using LT.DigitalOffice.ProjectService.Models.Dto.Requests;
-using LT.DigitalOffice.ProjectService.Validation.Interfaces;
+using LT.DigitalOffice.ProjectService.Validation.User.Interfaces;
 using MassTransit;
 using Microsoft.Extensions.Logging;
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs:21]: Rename to CreateProjectUsersCommand

File: src/ProjectService.Business/Commands/Project/GetProjectCommand.cs [hunk 20]
```diff
@@ -323,19 +320,19 @@ private async Task<(List<DepartmentData> departments, List<PositionData> positio
...
 
-    public async Task<OperationResultResponse<ProjectResponse>> Execute(GetProjectFilter filter)
+    public async Task<OperationResultResponse<ProjectResponse>> ExecuteAsync(GetProjectFilter filter)
     {
       OperationResultResponse<ProjectResponse> response = new();
       DepartmentData department = null;
-      DbProject dbProject = _repository.Get(filter);
+      DbProject dbProject = await _repository.GetAsync(filter);
 
       if (dbProject.DepartmentId.HasValue)
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Project/GetProjectCommand.cs:329]: we don't have department service yet

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs [hunk 21]
```diff
@@ -27,30 +29,32 @@ public class AddUsersToProjectCommand : IAddUsersToProjectCommand
     private readonly IAddUsersToProjectValidator _validator;
     private readonly ILogger<AddUsersToProjectCommand> _logger;
     private readonly IRequestClient<ICreateWorkTimeRequest> _rcCreateWorkTime;
+    private readonly IResponseCreater _responseCreater;
 
-    // TODO make async
-    private void CreateWorkTime(Guid projectId, List<Guid> userIds, List<string> errors)
+    private async Task CreateWorkTimeAsync(Guid projectId, List<Guid> userIds, List<string> errors)
     {
       string errorMessage = $"Failed to create a work time for project {projectId} with users: {string.Join(", ", userIds)}.";
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs:36]: you can remove it in 57 line

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs (hunk 11 above)
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: fixed

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs (hunk 12 above)
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: fixed

File: src/ProjectService.Business/Commands/Image/Interfaces/IRemoveImageCommand.cs [hunk 22]
```diff
@@ -7,6 +8,6 @@ namespace LT.DigitalOffice.ProjectService.Business.Commands.Image.Interfaces
   [AutoInject]
   public interface IRemoveImageCommand
   {
-    OperationResultResponse<bool> Execute(RemoveImageRequest request);
+    Task<OperationResultResponse<bool>> Execute(RemoveImageRequest request);
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Image/Interfaces/IRemoveImageCommand.cs]: fixed

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 23]
```diff
@@ -36,20 +36,20 @@ public class CreateImageCommand : ICreateImageCommand
     private readonly IDbImageMapper _dbProjectImageMapper;
     private readonly ICreateImageValidator _validator;
     private readonly IUserRepository _userRepository;
-    private readonly ITaskRepository _taskRepository;
 
     private List<Guid> CreateImages(List<ImageContent> context, Guid userId, Guid enityId, List<string> errors)
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: fixed

File: src/ProjectService.Business/Commands/ProjectUsers/RemoveProjectUsersCommand.cs [hunk 24]
```diff
@@ -1,44 +1,47 @@
...
+      _responseCreater = responseCreater;
     }
 
-    // TODO make async
-
-    public OperationResultResponse<bool> Execute(Guid projectId, List<Guid> userIds)
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid projectId, List<Guid> userIds)
     {
-      if (userIds == null || !userIds.Any())
+      if (!(await _accessValidator.HasRightsAsync(Rights.AddEditRemoveProjects)))
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/ProjectUsers/RemoveProjectUsersCommand.cs:34]: project admin can add  and remove users too

File: src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs (hunk 19 above)
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/ProjectUsers/AddUsersToProjectCommand.cs]: fixed

File: src/ProjectService.Data/ProjectRepository.cs [hunk 25]
```diff
@@ -19,85 +19,147 @@ public class ProjectRepository : IProjectRepository
...
+      await _provider.SaveAsync();
 
       return dbProject.Id;
     }
 
-    public bool Edit(Guid projectId, JsonPatchDocument<DbProject> request)
+    public async Task<bool> EditAsync(Guid projectId, JsonPatchDocument<DbProject> request)
     {
-      DbProject dbProject = Get(new GetProjectFilter { ProjectId = projectId });
+      DbProject dbProject = await _provider.Projects.FirstOrDefaultAsync(p => p.Id == projectId);
```
Comment (MariyaSin) [inline src/ProjectService.Data/ProjectRepository.cs:148]: Could you check if dbProject == default?

File: src/ProjectService.Business/Commands/ProjectUsers/RemoveProjectUsersCommand.cs (hunk 24 above)
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/ProjectUsers/RemoveProjectUsersCommand.cs]: fixed

File: src/ProjectService.Data/Interfaces/IUserRepository.cs [hunk 26]
```diff
@@ -14,39 +15,22 @@ namespace LT.DigitalOffice.ProjectService.Data.Interfaces
...
 
-    bool AreUserProjectExist(Guid userId, Guid projectId, bool? isManager = null);
+    Task<List<DbProjectUser>> FindAsync(FindDbProjectsUserFilter filter);
 
-    /// <summary>
-    /// Check that users are exist.
-    /// </summary>
-    /// <param name="ids">Ids to check that all of them exists.</param>
-    bool AreExist(params Guid[] ids);
+    Task<List<DbProjectUser>> GetAsync(List<Guid> userIds);
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Data/Interfaces/IUserRepository.cs]: fixed

File: src/ProjectService.Data/Interfaces/IUserRepository.cs [hunk 27]
```diff
@@ -14,39 +15,22 @@ namespace LT.DigitalOffice.ProjectService.Data.Interfaces
...
-    void Remove(Guid userId, Guid removedBy);
+    Task<List<Guid>> DoExistAsync(Guid projectId, List<Guid> ids);
 
-    List<DbProjectUser> Get(IGetProjectsUsersRequest request, out int totalCount);
+    Task<bool> RemoveAsync(Guid userId, Guid removedBy);
 
-    bool DisableWorkersInProject(Guid projectId, IEnumerable<Guid> userIds);
-
-    List<Guid> GetExistProjectUsers(Guid projectId, IEnumerable<Guid> userIds);
+    Task<bool> RemoveAsync(Guid projectId, IEnumerable<Guid> userIds);
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Data/Interfaces/IUserRepository.cs]: fixed

File: src/ProjectService.Data/ProjectRepository.cs (hunk 25 above)
Comment (BelehovEgor) [reply src/ProjectService.Data/ProjectRepository.cs:143]: fixed

File: src/ProjectService.Data/UserRepository.cs [hunk 28]
```diff
@@ -50,83 +77,70 @@ public IEnumerable<DbProjectUser> GetProjectUsers(Guid projectId, bool showNotAc
...
-
-      _provider.ProjectsUsers.AddRange(dbProjectUsers);
-      _provider.Save();
-
-      return true;
+      return await _provider.ProjectsUsers.Where(u => userIds.Contains(u.UserId) && u.IsActive).ToListAsync();
     }
 
-    public IEnumerable<DbProjectUser> Find(Guid userId)
+    public async Task<List<Guid>> GetExistAsync(Guid projectId, IEnumerable<Guid> userIds)
```
Comment (MariyaSin) [inline src/ProjectService.Data/UserRepository.cs]: usersIds

File: src/ProjectService.Validation/Image/CreateImageValidator.cs [hunk 29]
```diff
@@ -1,26 +1,27 @@
...
     {
       List<string> errors = new();
 
       RuleFor(images => images)
         .NotNull().WithMessage("List must not be null.")
         .NotEmpty().WithMessage("List must not be empty.");
 
-      RuleFor(images => images.EntityId)
+      RuleFor(images => images.ProjectId)
         .NotEmpty().WithMessage("Image's Id must not be empty.");
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/CreateImageValidator.cs:20]: Project id must....

File: src/ProjectService.Mappers/Responses/Interfaces/IProjectResponseMapper.cs [hunk 30]
```diff
@@ -1,15 +1,14 @@
...
 using System.Collections.Generic;
 
 namespace LT.DigitalOffice.ProjectService.Mappers.Responses.Interfaces
 {
-    [AutoInject]
+  [AutoInject]
     public interface IProjectResponseMapper
     {
-        ProjectResponse Map(DbProject dbProject, IEnumerable<ProjectUserInfo> users, IEnumerable<ProjectFileInfo> files, IEnumerable<ImageInfo> images, DepartmentInfo department);
+        ProjectResponse Map(DbProject dbProject, IEnumerable<UserInfo> users, IEnumerable<ProjectFileInfo> files, IEnumerable<ImageInfo> images, DepartmentInfo department);
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Mappers/Responses/Interfaces/IProjectResponseMapper.cs]: fixed

File: src/ProjectService.Validation/Image/CreateImageValidator.cs (hunk 29 above)
Comment (BelehovEgor) [reply src/ProjectService.Validation/Image/CreateImageValidator.cs]: fixed

File: src/ProjectService/Controllers/UserController.cs [hunk 31]
```diff
@@ -11,21 +12,21 @@ namespace LT.DigitalOffice.ProjectService.Controllers
...
     {
-      return command.Execute(request);
+      return await command.ExecuteAsync(request);
     }
 
-    [HttpDelete("removeUsersFromProject")]
-    public OperationResultResponse<bool> RemoveUsersFromProject(
-      [FromServices] IRemoveUsersFromProjectCommand command,
+    [HttpDelete("remove")]
+    public async Task<OperationResultResponse<bool>> Remove(
```
Comment (MariyaSin) [inline src/ProjectService/Controllers/UserController.cs:24]: RemoveAsync

File: src/ProjectService/Controllers/ProjectController.cs [hunk 32]
```diff
@@ -20,32 +20,32 @@ public class ProjectController : ControllerBase
...
     public async Task<OperationResultResponse<ProjectResponse>> Get(
       [FromServices] IGetProjectCommand command,
       [FromQuery] GetProjectFilter filter)
     {
-      return await command.Execute(filter);
+      return await command.ExecuteAsync(filter);
     }
 
     [HttpPost("create")]
     public async Task<OperationResultResponse<Guid?>> Create(
```
Comment (MariyaSin) [inline src/ProjectService/Controllers/ProjectController.cs:35]: CreateAsync

File: src/ProjectService.Validation/Project/CreateProjectRequestValidator.cs [hunk 33]
```diff
@@ -98,7 +99,7 @@ private async Task<bool> CheckValidityDepartmentId(Guid? departmentId)
         return true;
       }
 
-      string logMessage = "Department with id: {id} not found.";
+      var logMessage = "Department with id: {id} not found.";
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Project/CreateProjectRequestValidator.cs:102]: do not use var

Comment (BelehovEgor) [reply src/ProjectService.Validation/Project/CreateProjectRequestValidator.cs]: fixed

File: src/ProjectService/Controllers/ProjectController.cs (hunk 32 above)
Comment (BelehovEgor) [reply src/ProjectService/Controllers/ProjectController.cs]: fixed

File: src/ProjectService/Controllers/UserController.cs [hunk 34]
```diff
@@ -11,21 +12,21 @@ namespace LT.DigitalOffice.ProjectService.Controllers
...
+      [FromServices] IRemoveProjectUsersCommand command,
       [FromQuery] Guid projectId,
       [FromBody] List<Guid> userIds)
     {
-      return command.Execute(projectId, userIds);
+      return await command.ExecuteAsync(projectId, userIds);
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService/Controllers/UserController.cs]: fixed

**Review by Cheephot — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**